src/gpu/nvidia/cudnn_matmul_executor.hpp

```diff
@@ -392,16 +392,16 @@ struct cudnn_matmul_lt_exec_t final : public cudnn_matmul_lt_base_exec_t {
                             memory_tracking::names::key_matmul_dst_in_acc_dt)
                     : xpu::sycl::interop_memory_arg_t<
                             ::sycl::access::mode::read_write>();
-            auto arg_block_a_scratch = params->source_size_ != 0
```

Comment:
Yes.

>  If so, should we consider adding more tests to catch them?

From what I can tell, this file should be covering all the possible combinations, and I do see the case for which this was failing (i.e A = S8, ab format, B = S8, ba format, and C = S32, ab format), but maybe others with a better understanding could chime in, maybe @ShanoToni ?

Reply:
This is an oversight on the original PR, caused by differences in matmuls (oneDNN is col major, cuDNN is row major so we do BxA instead of AxB) the `arg_block_a_scratch` should be mapped to weights to be correct, so in the case where we need scratchpad memory only for weights and not source we would get correctness errors.

This is difficult to add more tests as cublasLT would only run when other implementations do not support it (cublas supports most already), additionally I am not sure what would the process for expanding the tests being run.

Reply:
Thanks for the explanation. I suppose it's fine then.